### PR TITLE
Implement get objects or log

### LINKED ISSUE #4803: move_files tasks depends on the Version from the db and fails
Similar to #4789, we pass the `Version.pk` to the `move_files` task. If the version is removed before the task is executed, the task fails because the row is not in the database anymore.

The problem is at this line,

https://github.com/rtfd/readthedocs.org/blob/3fbb0c1a7c94bcc979f54d67fad8dc94f74854c8/readthedocs/projects/tasks.py#L838

We should handle this exception properly and do not fail the task. We need to think this a little more, but I suppose that if there is no version in the database, there is nothing to move or do at that point. So, I think that just adding a `try/except` block with an `log.warning` will do the trick.

This happens frequently: https://sentry.io/read-the-docs/readthedocs-org/issues/235379529/?query=is:unresolved

### PR DESCRIPTION
**Pull request by dojutsu-user**
Fixes #4803

### CODE REVIEW

**Comment by dojutsu-user** [comment]
@humitos 
Please review.

**Comment by agjohnson** [comment]
@dojutsu-user Just a heads up, requesting review directly from someone on core isn't necessary, someone will review the work when time allows.

**Comment by dojutsu-user** [comment]
@agjohnson 
I will take care of it in the future.

**Comment by dojutsu-user** [comment]
I have added the tests, it is ready for review.

File: readthedocs/builds/managers.py [hunk 1]
```diff
@@ -67,6 +72,12 @@ def create_latest(self, **kwargs):
         defaults.update(kwargs)
         return self.create(**defaults)
 
+    def get_object_or_log(self, **kwargs):
+        try:
+            return super(VersionManagerBase, self).get(**kwargs)
```
Comment (humitos) [inline readthedocs/builds/managers.py]: We are running py3 only. We should replace this with just `super().get`

File: readthedocs/builds/managers.py [hunk 2]
```diff
@@ -67,6 +72,12 @@ def create_latest(self, **kwargs):
...
+        except ObjectDoesNotExist:
+            log.warning('Version not found for the pk = {pk}'.format(pk=kwargs.get('pk')))
```
Comment (humitos) [inline readthedocs/builds/managers.py]: We should use `.format` when logging and should use `%s` instead.

Also, this method allows us to be called with different attributes than just `pk`. We should log all the ones used here.

File: readthedocs/builds/managers.py [hunk 3]
```diff
@@ -67,6 +72,12 @@ def create_latest(self, **kwargs):
         defaults.update(kwargs)
         return self.create(**defaults)
 
+    def get_object_or_log(self, **kwargs):
```
Comment (humitos) [inline readthedocs/builds/managers.py]: Please, add a docstring for this method so we know how it's called, what it returns and where or when we call it.

File: readthedocs/builds/managers.py (hunk 1 above)
Comment (dojutsu-user) [reply readthedocs/builds/managers.py]: Thank you.
I have pushed the changes.

File: readthedocs/builds/managers.py (hunk 2 above)
Comment (dojutsu-user) [reply readthedocs/builds/managers.py]: Thank you.
Learned something new.
I have pushed the changes.

**Comment by ericholscher** [comment]
:+1:

**Pull request merged**